app/jobs/signal_adapter/set_username_job.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'net/http'
+
+module SignalAdapter
+  class SetUsernameJob < ApplicationJob
+    def perform(organization_id:)
+      organization = Organization.find(organization_id)
+      uri = URI.parse("#{ENV.fetch('SIGNAL_CLI_REST_API_ENDPOINT', 'http://localhost:8080')}/v1/accounts/#{organization.signal_server_phone_number}/username")
+      request = Net::HTTP::Post.new(uri, {
+                                      Accept: 'application/json',
+                                      'Content-Type': 'application/json'
+                                    })
+      request.body = { username: organization.project_name.gsub(/[^\w\s]/, '').gsub(/\s+/, '').camelize }.to_json
```

Comment:
Require a project name for every organization. The string must be validated according to signal's validations.